src/renderer/pages/Staking/Overview/components/NominatorsList/ShardedList/ShardedList.tsx

```diff
@@ -0,0 +1,201 @@
+import { Fragment, ReactNode, useEffect, useMemo, useState } from 'react';
+import { Transition } from '@headlessui/react';
+import { Trans } from 'react-i18next';
+
+import { useI18n } from '@app/providers';
+import type { Asset, Explorer, Address, NominatorInfo } from '@shared/core';
+import { cnTw } from '@shared/lib/utils';
+import { FootnoteText, Checkbox, Accordion, Shimmering, Plate, Tooltip } from '@shared/ui';
+import { ShardAccount } from '@shared/core/types/account';
+import { DotStyles } from '@shared/ui/StatusLabel/common/constants';
+import { AssetBalance } from '@entities/asset';
+import { AssetFiatBalance } from '@entities/price/ui/AssetFiatBalance';
+import { useStakingData } from '@entities/staking';
+import { NominatorsItem } from '../NominatorItem/NominatorItem';
+
+type NominatorInfoWithShard = NominatorInfo<ShardAccount>;
+
+type Props = {
+  shardsStake: NominatorInfoWithShard[];
+  era?: number;
+  asset?: Asset;
+  explorers?: Explorer[];
+  isStakingLoading: boolean;
+  onCheckValidators: (stash?: Address) => void;
+  onToggleNominator: (nominator: Address, value?: boolean) => void;
+  getContent: (stake: NominatorInfoWithShard) => ReactNode;
+};
+
+export const ShardedList = ({
+  shardsStake,
+  era,
+  asset,
+  explorers,
+  isStakingLoading,
+  onCheckValidators,
+  onToggleNominator,
+  getContent,
+}: Props) => {
+  const { getNextUnstakingEra, hasRedeem } = useStakingData();
+
+  const { t } = useI18n();
+  const [shards, setShards] = useState<NominatorInfoWithShard[]>(shardsStake);
+  const [selectedShards, setSelectedShards] = useState<number>(0);
+  const [withdrawnShards, setWithdrawnShards] = useState<number>(0);
+  const [unstakingShards, setUnstakingShards] = useState<number>(0);
+
+  useEffect(() => {
+    setShards(shardsStake);
+  }, [shardsStake]);
+
+  const selectAllShards = (value: boolean) => {
+    const allShards = shards.map((shard) => {
+      onToggleNominator(shard.address, value);
+      setSelectedShards(value ? shards.length : 0);
```

Comment:
`setSelectedShards(value ? shards.length : 0);` it doesn't use `shard` from `shards.map`
why it stays inside loop?